src/api/interactivity.js

```diff
@@ -134,14 +135,26 @@ export default class Interactivity {
         return this._emitter.off(eventName, callback);
     }
 
-    _init(layerList) {
+    _init(layerList, options) {
         this._emitter = mitt();
         this._layerList = layerList;
         this._prevHoverFeatures = [];
         this._prevClickFeatures = [];
+        this._numListeners = {};
         return Promise.all(layerList.map(layer => layer._context)).then(() => {
             postCheckLayerList(layerList);
             this._subscribeToIntegratorEvents(layerList[0].getIntegrator());
+        }).then(() => {
+            if (!options.defaultCursor) {
+                this._setInteractiveCursor();
+            }
+        });
+    }
+
+    _setInteractiveCursor() {
+        this.on('featureHover', event => {
+            const map = this._layerList[0].getIntegrator().map; // All layers belong to the same map
+            map.getCanvas().style.cursor = event.features.length ? 'pointer' : '';
```

Comment:
If there are multiple interactivity objects for the same map and more than one interactivity enables the feature, I think it's gonna misbehave. They will race between each other if there is a feature for one of the interactivities but there is no feature for the other one.

Reply:
Instead of setting the cursor to `''` by default, I would set it to `'auto'`. 

About what @davidmanzanares said, I did not understand it very well. If there are multiple interactivity objects, at the end the only purpose of this function is to change the cursor pointer by checking if at least there is one feature, right? I am missing how it could fail :( 

Also, since all the layers belong to the same map, would it be possible to have a property 'map' so we can get the map when we need it?

Reply:
> I think it's gonna misbehave

I think there is no easy way to manage this but advising the users to use the brain and don't create crazy mixes x) 

In any case:

- x interactivities with the same config: no problem
- x interactivities with different configs:
  - no overlapping features: no problem
  - overlapping features: Callbacks will collide and the last executed will win

Reply:
> Instead of setting the cursor to '' by default, I would set it to 'auto'

We dont want the cursor to be auto, we want the cursor to be the one defined by mapbox (`grab`)

If we do `canvas.style.cursor = 'auto'` we are creating a css more specific css rule that overrides the one given by mapbox leading to a wrong behaviour.

When we do `canvas.style.cursor = ''` we are creating a empty specific rule so the browser applies the one given by mapbox.

Reply:
👍

Reply:
@davidmanzanares I don't understand your proposal

Reply:
1. Create a per-map `Set` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set ) initialized to empty. This set is going to keep track of the Interactivity objects that want to change the cursor to `pointer`.
2. Each time the hover callback is invoked do:
2.1 Check if there is a feature behind the cursor for its layers.
2.2 If there is a `hit`, then add the Interactivity object to the set, if not, remove the interactivity object from the set
2.3 Set the style cursor to `pointer` if the set object is not empty, set it to the default otherwise.